api/BaseClient.ts

```diff
@@ -1,11 +1,44 @@
 export default class BaseClient {
   constructor(protected baseUrl: URL | string) {}
 
-  protected async get<R>(path: string): Promise<R> {
-    const response = await fetch(new URL(path, this.baseUrl), {
-      method: 'GET',
-    });
+  protected async get<R>(path: string, options?: RequestInit): Promise<R> {
+    if (options) {
+      Object.assign(options, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
```

Comment:
`Content-Type` gibt den Typen des Bodies an, den wir hier nicht schicken.

```suggestion
          'Accept': 'application/json',
```

Oder einfach weglassen...